lib/pharos/kube.rb

```diff
@@ -10,61 +10,33 @@ module Kube
     autoload :Stack, 'pharos/kube/stack'
     autoload :Session, 'pharos/kube/session'
 
-    # @param host [String]
-    # @return [Kubeclient::Client]
-    def self.client(host, version = 'v1')
-      @kube_client ||= {}
-      unless @kube_client[version]
-        config = host_config(host)
-        path_prefix = version == 'v1' ? 'api' : 'apis'
-        api_version, api_group = version.split('/').reverse
-        @kube_client[version] = Pharos::Kube::Client.new(
-          (config.context.api_endpoint + "/#{path_prefix}/#{api_group}"),
-          api_version,
-          ssl_options: config.context.ssl_options,
-          auth_options: config.context.auth_options
-        )
-      end
-      @kube_client[version]
-    end
-
-    # @param host [String]
+    # @param host [Pharos::Configuration::Host]
     # @return [Pharos::Kube::Session]
     def self.session(host)
-      @sessions ||= {}
-      @sessions[host] ||= Session.new(host)
+      Session.new(host)
     end
 
-    # @param host [String]
+    # @param host [Pharos::Configuration::Host]
     # @return [Kubeclient::Config]
     def self.host_config(host)
       Kubeclient::Config.read(host_config_path(host))
     end
 
-    # @param host [String]
+    # @return [String]
+    def self.config_dir
+      File.join(Dir.home, '.pharos')
+    end
+
+    # @param host [Pharos::Configuration::Host]
     # @return [String]
     def self.host_config_path(host)
```

Comment:
Maybe this stuff should actually be in `Pharos::Config::Host`

Reply:
I dont't need to touch it here, we can refactor the config harder later if we want to.

Reply:
Also, the configs are not actually per-host... if you configure with `api.endpoint`, then you'll only have one config file for that endpoint... so it's a little illogical to actually involve the `Pharos::Configuration::Host` here, because it's not actually host-specific. Instead, the cluster config has an api endpoint that defaults to the address of the first master host...

Reply:
Flipped the `api_endpoint` logic around, it's now defined on the top-level config, not on the host.